Updated ses_config.yml file generation to DeepSea openstack.integrate format

matthew-fuller opened the pull request:
Modified ses_config.yml.j2 template and added a task to make the output ses_config.yml file from the ses-osh role match the DeepSea openstack.integrate output format. Tested and verified locally (KVM), but not on eng cloud...

matthew-fuller commented:
PR #113 addresses this problem (and accounts for a problem parsing the variable name 'cinder-backup' in the new format due to the hyphen).

evrardjp commented on playbooks/roles/ses-osh/tasks/main.yml:
```diff
@@ -11,6 +11,15 @@
+  set_fact:
+    "ceph_{{ item | regex_replace('[\\.\\-]', '_') }}_key":
+       "{{lookup('ini', 'key section={{ item }} file={{ socok8s_workspace }}/ceph.{{ item }}.keyring') }}"
+  with_items:
```
Shouldn't this be based on the ceph_files.results too, to avoid having an item list that's out of date, and ensuring everything is properly fetched?

evrardjp commented on playbooks/roles/ses-osh/tasks/main.yml:
```diff
@@ -11,6 +11,15 @@
 
+- name: Get key values for ses-config.yml
+  set_fact:
+    "ceph_{{ item | regex_replace('[\\.\\-]', '_') }}_key":
```
Where is this fact used? I am confused why this needs to be a set fact.

matthew-fuller replied:
I tried to leave most of the original PR (36) intact. These variables are used in the ses_config.yml.j2 template when constructing the new ses_config file, and they were originally based on the names of the ses_openstack_keys in the ses-ansible role, which are hard coded to "client.glance", "client.cinder", and "client.cinder_backup". However, those keys don't seem to be accessible from the ses-osh role, so I simply made a list of the names. I'm certainly open to changing / improving this, however.

evrardjp replied:
Oh yeah I see now. 

Wouldn't it be simpler to do the lookups directly in the template? It would be far easier to read/parse.

matthew-fuller commented:
I closed PR 113 and moved its changes into this PR since they all pertained to the new ses_config format. Per JP's request in PR 113, I changed the "Get ses_cluster_configuration" task in deploy-osh/tasks/main.yml to use include_vars rather than using filtering.

evrardjp commented on playbooks/roles/ses-osh/templates/ses-config.yml.j2:
```diff
@@ -1,28 +1,24 @@
+  mon_initial_members: {{ hostvars[groups['ses_nodes'][0]].conf_options.stdout_lines.1 }}
+  public_network: {{ hostvars[groups['ses_nodes'][0]].conf_options.stdout_lines.3 }}
+cinder:
+  key: {{ ceph_client_cinder_key }}
```
I meant with the comment above you can replace the set_fact with the lookup directly here, and avoid the messing up with the input/set fact :)

evrardjp commented:
Close to perfect to me. I'd remove the set fact for future reviewability

matthew-fuller commented:
Following JP's recommendation, I've removed the task "Get key values for ses-config" entirely and moved the client key lookups to the ses_config.yml.j2 template directly.

evrardjp commented:
I haven't tested it, but it's simple and looks great.

Itxaka commented:
merge?
